syntax = "proto3";

package user;

service UserService {
  // Registra un nuevo usuario
  rpc Register (RegisterRequest) returns (RegisterResponse);
  
  // Actualiza la contrase√±a de un usuario
  rpc UpdatePassword (UpdatePasswordRequest) returns (UpdatePasswordResponse);
}

message RegisterRequest {
  string name = 1;
  string surname = 2;
  string surname2 = 3;
  string email = 4;
  string password = 5;
}

message RegisterResponse {
  string message = 1;
}

message UpdatePasswordRequest {
  string email = 1;
  string current_password = 2;
  string new_password = 3;
}

message UpdatePasswordResponse {
  string message = 1;
}
